nannou/src/app.rs

```diff
@@ -749,6 +750,27 @@ impl App {
         windows.keys().cloned().collect()
     }
 
+    /// Return the **SwapChain** for the given `window_id`, rebuilding if it doesn't exist.
+    ///
+    /// **Panics** if `window_id` does not exist, or if the swap chain is missing and cannot be rebuilt.
+    pub fn get_or_rebuild_swap_chain(&self, window_id: window::Id) -> SwapChain {
```

Comment:
I wonder if a better name for this might be `take_or_rebuild_swap_chain`? I think `get` tends to be used more often when retrieving a reference rather than taking ownership.

Also, perhaps this method should be private?